src/server/game/AI/CoreAI/UnitAI.h

```diff
@@ -44,13 +45,15 @@ struct DefaultTargetSelector : public Acore::unary_function<Unit*, bool>
     Unit const* me;
     float m_dist;
     bool m_playerOnly;
+    Unit const* except;
     int32 m_aura;
 
     // unit: the reference unit
     // dist: if 0: ignored, if > 0: maximum distance to the reference unit, if < 0: minimum distance to the reference unit
     // playerOnly: self explaining
+    // withMainTank: allow current tank to be selected
     // aura: if 0: ignored, if > 0: the target shall have the aura, if < 0, the target shall NOT have the aura
-    DefaultTargetSelector(Unit const* unit, float dist, bool playerOnly, int32 aura) : me(unit), m_dist(dist), m_playerOnly(playerOnly), m_aura(aura) {}
+    DefaultTargetSelector(Unit const* unit, float dist, bool playerOnly, bool withMainTank, int32 aura) : me(unit), m_dist(dist), m_playerOnly(playerOnly), except(withMainTank ? me->GetThreatMgr().GetCurrentVictim() : nullptr), m_aura(aura) {}
```

Comment:
Just out of curiousity, will `me->GetThreatMgr().GetCurrentVictim()` ever return anything other than a `nullptr` here?